src/main/java/com/balako/onlinebookstore/mapper/CartItemMapper.java

```diff
@@ -0,0 +1,19 @@
+package com.balako.onlinebookstore.mapper;
+
+import com.balako.onlinebookstore.config.MapperConfig;
+import com.balako.onlinebookstore.dto.cart.response.CartItemDto;
+import com.balako.onlinebookstore.model.CartItem;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
+public interface CartItemMapper {
+    CartItemDto toDto(CartItem cartItem);
+
+    @AfterMapping
+    default void setBookTitleAndBookId(@MappingTarget CartItemDto cartItemDto, CartItem cartItem) {
+        cartItemDto.setBookTitle(cartItem.getBook().getTitle());
+        cartItemDto.setBookId(cartItem.getBook().getId());
+    }
```

Comment:
Or with annotation: 

```suggestion
    @Mapping(target = "bookId", source = "item.book.id")
    @Mapping(target = "bookTitle", source = "item.book.title")
    CartItemDto toDto(CartItem cartItem);
```

Reply:
I think using annotations will be preferable